lib/common/options.c

```diff
@@ -987,6 +987,85 @@ add_desc(xmlNode *parent, const char *tag, const char *desc)
     return pcmk_rc_ok;
 }
 
+/*!
+ * \internal
+ * \brief Add a \c PCMK_XE_PARAMETER element to an OCF-like metadata XML node
+ *
+ * \param[in,out] parent  Parent \c PCMK_XE_PARAMETERS node
+ * \param[in]     option  Option to add as a \c PCMK_XE_PARAMETER element
+ *
+ * \return Standard Pacemaker return code
+ */
+static int
+add_option_metadata(xmlNode *parent, const pcmk__cluster_option_t *option)
+{
+    const char *desc_long = option->description_long;
+    const char *desc_short = option->description_short;
+    xmlNode *parameter = NULL;
+    xmlNode *content = NULL;
+    int rc = pcmk_rc_ok;
+
+    // The standard requires long and short parameter descriptions
+    CRM_ASSERT((desc_long != NULL) || (desc_short != NULL));
+
+    if (desc_long == NULL) {
+        desc_long = desc_short;
+    } else if (desc_short == NULL) {
+        desc_short = desc_long;
+    }
+
+    // The standard requires a parameter type
+    CRM_ASSERT(option->type != NULL);
```

Comment:
Since we're returning and checking rich error codes, just return EINVAL instead of the two assertions

Reply:
It's a programming error since we're in control of the metadata, so I don't think it would hurt to catch it via an assert... though I can see an argument for not wanting to assert within a daemon, even if it is a soon-to-be-deprecated metadata call.

FWIW, in an upcoming commit (72a254a8), we no longer return rich error codes. It doesn't make sense to NULL-check created XML nodes and return `ENOMEM`, since the XML output functions would've segfaulted by then. So most of the error checking is dropped as of right now.